libs/mpi/include/hpx/mpi/mpi_future.hpp

```diff
@@ -0,0 +1,186 @@
+//  Copyright (c) 2019 John Biddiscombe
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_LIBS_MPI_FUTURE_HPP
+#define HPX_LIBS_MPI_FUTURE_HPP
+
+#include <mpi.h>
```

Comment:
Please #include HPX headers first (at least `#include <hpx/config.hpp>`) to avoid issues with Windows based compilations.